src/Workleap.OpenApi.MSBuild/msbuild/tools/Workleap.OpenApi.MSBuild.targets

```diff
@@ -14,15 +14,23 @@
   <!-- This is the location of this task assembly when debugging the task in an IDE -->
   <UsingTask Condition="'$(OpenApiDebuggingEnabled)' == 'true'" TaskName="$(MSBuildThisFileName).ValidateOpenApiTask" AssemblyFile="$(MSBuildThisFileDirectory)..\..\bin\$(Configuration)\netstandard2.0\$(MSBuildThisFileName).dll" />
 
-  <!-- We can only analyze the OpenAPI spec after the project has been built -->
-  <Target Name="ValidateOpenApi" DependsOnTargets="ResolveProjectReferences" AfterTargets="Build" Condition="'$(OpenApiEnabled)' == 'true'">
+  <!-- We validate and regenerate OpenAPI specs after the build stage. -->
+  <!-- This task is executed only if the compiled outputs are newer than the last task execution. -->
```

Comment:
We should add the source of this logic something like that so we can more easily find out next time

To find out if we need to validate open api we compared a touched files strategy with the output of the CoreCompile target and the Projects files